tests/linearizability/model.go

```diff
@@ -105,92 +100,99 @@ var etcdModel = porcupine.Model{
 	},
 }
 
-func step(state EtcdState, request EtcdRequest, response EtcdResponse) (bool, EtcdState) {
-	if len(state.PossibleValues) == 0 {
-		state.Key = request.Key
-		if ok, v := initValueRevision(request, response); ok {
-			state.PossibleValues = append(state.PossibleValues, v)
+func step(states []EtcdState, request EtcdRequest, response EtcdResponse) (bool, []EtcdState) {
+	if len(states) == 0 {
+		if ok, v := initState(request, response); ok {
+			states = append(states, v)
 		}
-		return true, state
-	}
-	if state.Key != request.Key {
-		panic("multiple keys not supported")
+		return true, states
 	}
 	if response.Err != nil {
-		for _, v := range state.PossibleValues {
-			newV, _ := stepValue(v, request)
-			state.PossibleValues = append(state.PossibleValues, newV)
+		for _, s := range states {
+			newState, _ := stepState(s, request)
+			states = append(states, newState)
 		}
 	} else {
 		var i = 0
-		for _, v := range state.PossibleValues {
-			newV, expectedResponse := stepValue(v, request)
+		for _, s := range states {
+			newState, expectedResponse := stepState(s, request)
 			if expectedResponse == response {
-				state.PossibleValues[i] = newV
+				states[i] = newState
 				i++
 			}
 		}
-		state.PossibleValues = state.PossibleValues[:i]
+		states = states[:i]
```

Comment:
clever idea with the state reduction.

Reply:
Yea, but looks like it's not very readable. Rewrote it to make the code cleaner.